tools/cldr-code/src/main/java/com/ibm/icu/dev/test/UnicodeKnownIssues.java

```diff
@@ -0,0 +1,140 @@
+package com.ibm.icu.dev.test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Map.Entry;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * “Known issues” manager.
+ * Intended to be shared between ICU, CLDR, &c.
+ * Test frameworks can create an instance of this to manage known issues
+ */
+public class UnicodeKnownIssues {
+    private Map<String, List<String>> knownIssues = new TreeMap<>();
+    /**
+     * Max number of lines to show by default (including the "more")
+     * unless -allKnownIssues is given. Must be at least 2.
+     */
+    public static final int KNOWN_ISSUES_CURTAILMENT = 2;
+
+    /**
+     * true if all issues should be shown, false if they should
+     * be curtailed.
+     */
+    private boolean allKnownIssues;
+
+    /**
+     * Construct a known issue manager
+     * @param allKnownIssues true if all known issues should be printed,
+     * not curtailed
+     */
+    public UnicodeKnownIssues(boolean allKnownIssues) {
+        this.allKnownIssues = allKnownIssues;
+    }
+
+    /**
+     * Base URL for browsing Unicode JIRA
+     */
+    public static final String UNICODE_JIRA_BROWSE = "https://unicode-org.atlassian.net/browse/";
+
+    static final Pattern ICU_TICKET_PATTERN = Pattern.compile(
+        "(?i)(?:icu-)?(\\d+)"
+    );
+    static final Pattern CLDR_TICKET_PATTERN = Pattern.compile(
+        "(?i)cldr(?:bug:)?(?:-)?(\\d+)"
+    );
+
+    /**
+     * Match all linkable ticket patterns
+     * @see {org.unicode.cldr.util.CLDRURLS#CLDR_TICKET_BROWSE}
+     */
+    static final Pattern UNICODE_JIRA_PATTERN = Pattern.compile(
+        "(CLDR|ICU)-(\\d+)"
+    );
+
+    /**
+     * Log the known issue.
+     * Call this from the test framework when logKnownIssue() is called.
+     *
+     * @param path Path to the error, will be returned in the
+     * known issue list
+     * @param ticket A ticket number string. For an ICU ticket, use "ICU-10245".
+     * For a CLDR ticket, use "CLDR-12345".
+     * For compatibility, "1234" -> ICU-1234 and "cldrbug:456" -> CLDR-456
+     * @param comment Additional comment, or null
+     *
+     */
+    public void logKnownIssue(String path, String ticket, String comment) {
+
+        StringBuffer descBuf = new StringBuffer(path);
+
+        if (comment != null && comment.length() > 0) {
+            descBuf.append(" (" + comment + ")");
+        }
+        String description = descBuf.toString();
+
+        String ticketLink = "Unknown Ticket";
+        if (ticket != null && ticket.length() > 0) {
```

Comment:
```suggestion
        if (!Strings.isNullOrEmpty(ticket)) {
```
https://guava.dev/releases/19.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String)

Reply:
is that available for ICU also?

Reply:
I was trying to keep the dependencies small here, seeing as how it already needs a hack to support java 1.7

Reply:
i don't see any other use of Strings besides one call to Strings.repeat(), is this a commonly used function?

Reply:
These things are common in Google code :-)
But good call on keeping it easy to sync into ICU where we are still on Java 7 (working on it...) and don't use Guava.